src/lean_dojo/interaction/dojo.py

```diff
@@ -190,7 +190,7 @@ def uses_lean4(self) -> bool:
     def uses_lean3(self) -> bool:
         return self.repo.uses_lean3
 
-    def __enter__(self) -> Tuple["Dojo", State]:
+    def __enter__(self) -> Tuple["Dojo", Union[State, CommandState]]:
```

Comment:
`State = Union[CommandState, TacticState]`. So we don't need to add `CommandState` here?